### PR TITLE
Support symbolic tracing of `SchNet`

### PR DESCRIPTION
**Pull request by hatemhelal**
This PR removes/modifies a couple control flow patterns that were used in the `SchNet` implementation that are not traceable with `torch.fx`:

* removing the assertion in the forward pass
* Replaced calling `scatter` directly with an `Aggregation` module

Without these changes, attempting symbolic tracing:
```
from torch_geometric.nn.fx import Transformer
from torch_geometric.nn import SchNet

model = SchNet()
tx = Transformer(model, debug=True)
tx.transform()
```

fails with:
```
TraceError: symbolically traced variables cannot be used as inputs to control flow
```

### CODE REVIEW

File: torch_geometric/nn/models/schnet.py [hunk 1]
```diff
@@ -112,9 +112,9 @@ def __init__(
         self.num_interactions = num_interactions
         self.num_gaussians = num_gaussians
         self.cutoff = cutoff
-        self.readout = readout
+        self.aggr = aggr_resolver(readout)
+        self.sum_aggr = SumAggregation()
         self.dipole = dipole
-        self.readout = 'add' if self.dipole else self.readout
```
Comment (lightaime) [inline torch_geometric/nn/models/schnet.py:117]: Could you clarify this change? Shouldn't we resolve the aggr of `readout` after this?

**Review by lightaime — COMMENTED** [review]
Thanks! LGTM.

File: torch_geometric/nn/models/schnet.py (hunk 1 above)
Comment (hatemhelal) [reply torch_geometric/nn/models/schnet.py:117]: yes, good spot! I'll fix this soon

**Review by lightaime — APPROVED** [review]
Cool. Thanks!

**Review by rusty1s — APPROVED** [review]

**Pull request merged**